src/mame/includes/fmtowns.h

```diff
@@ -187,6 +191,7 @@ class towns_state : public driver_device
 	required_device<upd71071_device> m_dma_1;
 	required_device<cdrom_image_device> m_cdrom;
 	required_device<cdda_device> m_cdda;
+	optional_device<fmt_scsi_slot_device> m_scsi_slot;
```

Comment:
I’d rather not have optional things added to the driver class.  If possible, it would be better to make a derived class for the first-generation machines, and move anything specific to them there.